doc/lua_api.txt

```diff
@@ -8930,8 +8943,307 @@ Used by `minetest.add_particlespawner`.
+            -- underlying type of the property. for now, all but the first and last
+            -- frame are ignored
+
+            -- frames
+
+                -- floats
+                0, 0,
+
+                -- vec3s
+                vector.new(0,0,0),
+                vector.new(0,0,0),
+
+                -- vec3 ranges
+                { min = vector.new(0,0,0), max = vector.new(0,0,0), bias = 0 },
+                { min = vector.new(0,0,0), max = vector.new(0,0,0), bias = 0 },
+
+                -- mixed
+                0, { min = vector.new(0,0,0), max = vector.new(0,0,0), bias = 0 },
+        },
     }
 
+All of the properties that can be defined in this way are listed in the next
+section, along with the datatypes they accept.
+
+#### List of particlespawner properties
+All of the properties in this list can be animated with `*_tween` tables
+unless otherwise specified. For example, `jitter` can be tweened by setting
+a `jitter_tween` table instead of (or in addition to) a `jitter` table/value.
+Types used are defined in the previous section.
+
+* vec3 range `pos`: the position at which particles can appear
+* vec3 range `vel`: the initial velocity of the particle
+* vec3 range `acc`: the direction and speed with which the particle
+  accelerates
+* vec3 range `jitter`: offsets the velocity of each particle by a random
+  amount within the specified range each frame. used to create Brownian motion.
+* vec3 range `drag`: the amount by which absolute particle velocity along
+  each axis is decreased per second.  a value of 1.0 means that the particle
+  will be slowed to a stop over the space of a second; a value of -1.0 means
+  that the particle speed will be doubled every second. to avoid interfering
+  with gravity provided by `acc`, a drag vector like `vector.new(1,0,1)` can
+  be used instead of a uniform value.
+* float range `bounce`: how bouncy the particles are when `collisiondetection`
+  is turned on. values less than or equal to `0` turn off particle bounce;
+  `1` makes the particles bounce without losing any velocity, and `2` makes
+  them double their velocity with every bounce.  `bounce` is not bounded but
+  values much larger than `1.0` probably aren't very useful.
+* float range `exptime`: the number of seconds after which the particle
+  disappears.
+* table `attract`: sets the birth orientation of particles relative to various
+  shapes defined in world coordinate space. this is an alternative means of
+  setting the velocity which allows particles to emerge from or enter into
+  some entity or node on the map, rather than simply being assigned random
+  velocity values within a range. the velocity calculated by this method will
+  be **added** to that specified by `vel` if `vel` is also set, so in most
+  cases **`vel` should be set to 0**. `attract` has the fields:
+  * string `kind`: selects the kind of shape towards which the particles will
```

Comment:
Here I propose to go with an code snippet instead of nested lists to improve readability and be consistent with other table definitions:

```
{

  kind = "point",  -- specifies the kind of shape towards which the particles will be oriented
  -- * "none":  no attractor is set and the `attractor` table is ignored
  -- * "point": the particles are attracted to a specific point in space.
  --            use this also if you want a sphere-like effect, in combination with
  --            the `radius` property.
  -- * "line":  the particles are attracted to an (infinite) line passing
  --            through the points `origin` and `angle`. use this for e.g. beacon
  --            effects, energy beam effects, etc.
  -- * "plane": the particles are attracted to an (infinite) plane on whose
  --            surface `origin` designates a point in world coordinate space. use this
  --            for e.g. particles entering or emerging from a portal.
  
  strength = { ... float range }, -- the speed with which particles will move towards the attractor.
  -- If negative, the particles will instead move away from that object.
  
  ...
}
```

Reply:
can you submit a PR with the changes you want to make to the documentation? i would like to avoid a drawn-out back-and-forth over this if possible :)